src/main/scala/com/gu/newsletterlistcleanse/UpdateBrazeUsersLambda.scala

```diff
@@ -31,7 +34,12 @@ class UpdateBrazeUsersLambda {
     logger.info(s"Starting $env")
     parseCleanseListSqsMessage(sqsEvent) match {
       case Right(cleanseLists) =>
-        process(cleanseLists)
+        val brazeResults = Await.result(process(cleanseLists), timeout)
+        brazeResults match {
+          case Left(brazeErrors)=>
+            brazeErrors.foreach(e => logger.error(s"Error updating Braze with Code ${e.code}: ${e.body}"))
+          case Right(_) => logger.info("Successfully updated Braze")
+        }
```

Comment:
Not sure I like this, but not sure what a better option would be

Reply:
what exactly is bothering you? is it the nested pattern match?
If so the solution is to extract the inner one into its own function.

However I'd argue both these matches are actually processing and may find their spot in the `process` method?

Reply:
Refactored it into a separate function for resolving the Braze results so that it doesn't have a nested pattern match